Failure on initial build (`npm cache hit undefined`)

Hi! I am trying to get started with this action. I have the job defined as follows:

```yaml
jobs:
  npm:
    name: Install/Cache Node Dependencies
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: bahmutov/npm-install@v1
        with:
          working-directory: frontend-server
```

But on the initial build it seems to think `undefined` as a cache key is a hit?

```bash
2020-10-21T18:44:31.0250591Z running npm-install GitHub Action
2020-10-21T18:44:31.0307580Z trying to restore cached NPM modules
2020-10-21T18:44:31.1088182Z npm cache hit undefined
2020-10-21T18:44:31.1089088Z installing NPM dependencies
2020-10-21T18:44:31.1119956Z npm at "/usr/local/bin/npm"
2020-10-21T18:44:31.1124089Z [command]/usr/local/bin/npm ci
2020-10-21T18:45:08.1532741Z npm ERR! code ENOENT
2020-10-21T18:45:08.1534647Z npm ERR! syscall open
2020-10-21T18:45:08.1537232Z npm ERR! path /home/runner/work/web/web/frontend-server/node_modules/npm/node_modules/is-date-object/package.json
2020-10-21T18:45:08.1554327Z npm ERR! errno -2
2020-10-21T18:45:08.1589285Z npm ERR! enoent ENOENT: no such file or directory, open '/home/runner/work/web/web/frontend-server/node_modules/npm/node_modules/is-date-object/package.json'
2020-10-21T18:45:08.1591026Z npm ERR! enoent This is related to npm not being able to find a file.
2020-10-21T18:45:08.1592548Z npm ERR! enoent 
2020-10-21T18:45:09.1393468Z 
2020-10-21T18:45:09.1397189Z npm ERR! A complete log of this run can be found in:
2020-10-21T18:45:09.1399714Z npm ERR!     /home/runner/.npm/_logs/2020-10-21T18_45_08_187Z-debug.log
2020-10-21T18:45:09.1676706Z Error: The process '/usr/local/bin/npm' failed with exit code 254
2020-10-21T18:45:09.1695647Z     at ExecState._setResult (/home/runner/work/_actions/bahmutov/npm-install/v1/dist/index.js:962:25)
2020-10-21T18:45:09.1697991Z     at ExecState.CheckComplete (/home/runner/work/_actions/bahmutov/npm-install/v1/dist/index.js:945:18)
2020-10-21T18:45:09.1700080Z     at ChildProcess.<anonymous> (/home/runner/work/_actions/bahmutov/npm-install/v1/dist/index.js:839:27)
2020-10-21T18:45:09.1701542Z     at ChildProcess.emit (events.js:210:5)
2020-10-21T18:45:09.1702855Z     at maybeClose (internal/child_process.js:1021:16)
2020-10-21T18:45:09.1704119Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
2020-10-21T18:45:09.1708308Z ##[error]The process '/usr/local/bin/npm' failed with exit code 254
```

Any idea if this is something I am doing wrong and might be able to resolve?

--- a/index.js
+++ b/index.js
@@ -213,7 +213,7 @@ const npmInstallAction = async () => {
     .map(s => s.trim())
     .filter(Boolean)
 
-  core.debug(`iterating over working ${workingDirectories.length} dis(s)`)
+  core.debug(`iterating over working ${workingDirectories.length} folder(s)`)
 
   for (const workingDirectory of workingDirectories) {
     await api.utils.installInOneFolder({ usePackageLock, workingDirectory })

--- a/index.js
+++ b/index.js
@@ -81,7 +81,7 @@ const install = (opts = {}) => {
   }
 
   const options = {
-    cwd: opts.workingDirectory
+    cwd: path.resolve(opts.workingDirectory)
   }
 
   if (shouldUseYarn) {
@@ -204,15 +204,16 @@ const npmInstallAction = async () => {
   const usePackageLock = getInputBool('useLockFile', true)
   core.debug(`usePackageLock? ${usePackageLock}`)
 
+  // Note: working directory for "actions/exec" should be absolute
+
   const wds = core.getInput('working-directory') || process.cwd()
+
   const workingDirectories = wds
     .split('\n')
     .map(s => s.trim())
     .filter(Boolean)
 
-  core.debug(
-    `iterating over working ${workingDirectories.length} directorie(s)`
-  )
+  core.debug(`iterating over working ${workingDirectories.length} dis(s)`)
 
   for (const workingDirectory of workingDirectories) {
     await api.utils.installInOneFolder({ usePackageLock, workingDirectory })

--- a/test/action-spec.js
+++ b/test/action-spec.js
@@ -12,7 +12,8 @@ const action = require('../index')
 const utils = action.utils
 
 describe('action', () => {
-  const cwd = '/path/to/mock/cwd'
+  // by resolving we normalize the folder on Linux and Windows CI
+  const cwd = path.resolve('/path/to/mock/cwd')
   const homedir = '/home/path/for/test/user'
 
   beforeEach(function() {

--- a/test/install-spec.js
+++ b/test/install-spec.js
@@ -2,6 +2,7 @@ const exec = require('@actions/exec')
 const core = require('@actions/core')
 const io = require('@actions/io')
 const quote = require('quote')
+const path = require('path')
 
 const action = require('../index')
 
@@ -15,6 +16,34 @@ describe('install command', () => {
 
   context('using Yarn', () => {
     const pathToYarn = '/path/to/yarn'
+
+    it('uses absolute working directory', async function() {
+      const opts = {
+        useYarn: true,
+        usePackageLock: true,
+        // only use relative path
+        workingDirectory: 'directory',
+        npmCacheFolder
+      }
+      sandbox
+        .stub(io, 'which')
+        .withArgs('yarn')
+        .resolves(pathToYarn)
+      sandbox
+        .stub(path, 'resolve')
+        .withArgs('directory')
+        .returns(workingDirectory)
+
+      await action.utils.install(opts)
+      expect(
+        this.exec,
+        'to use absolute working directory'
+      ).to.have.been.calledOnceWithExactly(
+        quote(pathToYarn),
+        ['--frozen-lockfile'],
+        { cwd: workingDirectory }
+      )
+    })
 
     it('and lock file', async function() {
       const opts = {
@@ -60,6 +89,33 @@ describe('install command', () => {
 
     beforeEach(function() {
       this.exportVariable = sandbox.stub(core, 'exportVariable')
+    })
+
+    it('uses absolute working directory', async function() {
+      const opts = {
+        useYarn: false,
+        usePackageLock: true,
+        // only use relative path
+        workingDirectory: 'directory',
+        npmCacheFolder
+      }
+      sandbox
+        .stub(io, 'which')
+        .withArgs('npm')
+        .resolves(pathToNpm)
+      sandbox
+        .stub(path, 'resolve')
+        .withArgs('directory')
+        .returns(workingDirectory)
+
+      await action.utils.install(opts)
+
+      expect(
+        this.exec,
+        'to use absolute working directory'
+      ).to.have.been.calledOnceWithExactly(quote(pathToNpm), ['ci'], {
+        cwd: workingDirectory
+      })
     })
 
     it('installs using lock file', async function() {

--- a/test/install-spec.js
+++ b/test/install-spec.js
@@ -11,7 +11,8 @@ describe('install command', () => {
     this.exec = sandbox.stub(exec, 'exec').resolves()
   })
 
-  const workingDirectory = '/current/working/directory'
+  // by resolving we normalize the folder on Linux and Windows CI
+  const workingDirectory = path.resolve('/current/working/directory')
   const npmCacheFolder = '/path/to/user/cache'
 
   context('using Yarn', () => {

